addons/account/models/account_tax.py

```diff
@@ -292,56 +286,99 @@ def _compute_is_used(self):
+
+        # Categorize the lines that were added/removed/modifying
+        existing_lines = []
+        new_lines = []
+        deleted_lines = []
+        for line in old_line_values_dict.copy():
+            val = old_line_values_dict.pop(line)
+            if line in new_line_values_dict:
+                existing_lines.append((line, val, new_line_values_dict.pop(line)))
+            else:
+                deleted_lines.append((line, _('Removed'), val))
+        for line, val in new_line_values_dict.items():
+            try:
+                # We don't want to log when users add a repartition line that is
+                # exactly similar to one he also deleted.
+                deleted_lines.remove((line, _('Removed'), val))
+            except ValueError:
+                new_lines.append((line, _('New'), val))
 
         kwargs = {}
-        for old_value, new_value in zip(old_values, new_values):
-            if old_value != new_value:
-                old_value = ast.literal_eval(old_value)
-                new_value = ast.literal_eval(new_value)
-                diff_keys = [key for key in old_value if old_value[key] != new_value[key]]
-                repartition_line = self.env['account.tax.repartition.line'].search([('id', '=', new_value['id'])])
+        for (document_type, sequence), old_value, new_value in existing_lines:
+            diff_keys = [key for key in old_value if old_value[key] != new_value[key]]
+            if diff_keys:
                 body = Markup("<b>{type}</b> {rep} {seq}:<ul class='mb-0 ps-4'>{changes}</ul>").format(
-                    type=repartition_line.document_type.capitalize(),
+                    type=document_type.capitalize(),
                     rep=_('repartition line'),
-                    seq=repartition_line.sequence + 1,
+                    seq=sequence,
                     changes=Markup().join(
                         [Markup("""
                             <li>
                                 <span class='o-mail-Message-trackingOld me-1 px-1 text-muted fw-bold'>{old}</span>
                                 <i class='o-mail-Message-trackingSeparator fa fa-long-arrow-right mx-1 text-600'/>
                                 <span class='o-mail-Message-trackingNew me-1 fw-bold text-info'>{new}</span>
                                 <span class='o-mail-Message-trackingField ms-1 fst-italic text-muted'>({diff})</span>
-                            </li>""").format(old=old_value[diff_key], new=new_value[diff_key], diff=diff_key)
+                            </li>
+                        """).format(old=old_value[diff_key], new=new_value[diff_key], diff=diff_key)
                         for diff_key in diff_keys]
                     )
                 )
                 kwargs['body'] = body
                 super()._message_log(**kwargs)
 
+        for (document_type, sequence), operation, value in new_lines + deleted_lines:
```

Comment:
```suggestion
        for (document_type, sequence), operation, value in new_lines + deleted_lines:
```
Do you still need to keep 2 separate lists?